### PR TITLE
MONIT-27976: Add Service and Application tags to the default tags

### PR DESCRIPTION
**Pull request by xuranchen**

### CODE REVIEW

**Comment by joannak-vmware** [comment]
3 tests skipped. Please address failed build for Java11
```
 Tests in error: 
2022-04-14T21:20:02.5991904Z   testNullGaugeDoesNotThrow(com.wavefront.integrations.metrics.WavefrontYammerHttpMetricsReporterSecondaryTest): client with id proxy://127.0.0.1:0 already exists.
2022-04-14T21:20:02.5993155Z   testNullGaugeDoesNotThrow(com.wavefront.integrations.metrics.WavefrontYammerHttpMetricsReporterSecondaryTest)
```

File: java-lib/src/main/java/com/wavefront/ingester/ReportHistogramIngesterFormatter.java [hunk 1]
```diff
@@ -35,10 +35,10 @@ public static IngesterFormatBuilder<ReportHistogram> newBuilder() {
...
   @Override
   public ReportHistogram drive(String input, Supplier<String> defaultHostNameSupplier,
-                           String customerId, @Nullable List<String> customSourceTags,
-                           @Nullable List<String> customLogTimestampTags,
-                           @Nullable List<String> customLogMessageTags,
-                           @Nullable IngesterContext ingesterContext) {
+                               String customerId, @Nullable List<String> customSourceTags,
+                               @Nullable List<String> customLogTimestampTags,
+                               @Nullable List<String> customLogMessageTags,
+                               List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportHistogramIngesterFormatter.java:41]: nit: reformat line, please shift ingesterContext to next line

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 2]
```diff
@@ -598,12 +596,86 @@ public static Long getLogTimestamp(@Nullable List<Annotation> annotations,
...
   public T drive(String input, @Nullable Supplier<String> defaultHostNameSupplier,
                  String customerId) {
-    return drive(input, defaultHostNameSupplier, customerId, null, null, null, null);
+    return drive(input, defaultHostNameSupplier, customerId, null, null, null, null, null, null);
   }
 
   public abstract T drive(String input, @Nullable Supplier<String> defaultHostNameSupplier,
                           String customerId, @Nullable List<String> customSourceTags, @Nullable List<String> customLogTimestampTags,
-                          @Nullable List<String> customLogMessageTags, @Nullable IngesterContext ingesterContext);
+                          @Nullable List<String> customLogMessageTags, List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext);
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:680]: nit: reformat, move ingesterContext to next line

**Review by amitw-vmware — APPROVED** [review]
LGTM

File: java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java [hunk 3]
```diff
@@ -40,7 +37,7 @@ public static IngesterFormatBuilder<ReportLog> newBuilder() {
     @Override
     public ReportLog drive(String logJson, @Nullable Supplier<String> defaultHostNameSupplier,
                            String customerId, @Nullable List<String> customSourceTags, @Nullable List<String> customLogTimestampTags,
-                           @Nullable List<String> customLogMessageTags, @Nullable IngesterContext ingesterContext) {
+                           @Nullable List<String> customLogMessageTags, List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportLogIngesterFormatter.java:40]: nit: ditto, format

File: java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java [hunk 4]
```diff
@@ -37,7 +37,7 @@ public Span drive(String input, @Nullable Supplier<String> defaultHostNameSuppli
                     String customerId, @Nullable List<String> customSourceTags,
                     @Nullable List<String> customLogTimestampTags,
                     @Nullable List<String> customLogMessageTags,
-                    @Nullable IngesterContext ingesterContext) {
+                    List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/SpanIngesterFormatter.java:40]: nit: format

File: java-lib/src/main/java/com/wavefront/ingester/ReportPointIngesterFormatter.java [hunk 5]
```diff
@@ -41,7 +41,7 @@ public ReportPoint drive(String input, Supplier<String> defaultHostNameSupplier,
                            String customerId, @Nullable List<String> customSourceTags,
                            @Nullable List<String> customLogTimestampTags,
                            @Nullable List<String> customLogMessageTags,
-                           @Nullable IngesterContext ingesterContext) {
+                           List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportPointIngesterFormatter.java:44]: nit format

File: java-lib/src/main/java/com/wavefront/ingester/ReportSourceTagIngesterFormatter.java [hunk 6]
```diff
@@ -29,7 +29,7 @@ public ReportSourceTag drive(String input, @Nullable Supplier<String> defaultHos
                                String customerId, @Nullable List<String> customerSourceTags,
                                @Nullable List<String> customLogTimestampTags,
                                @Nullable List<String> customLogMessageTags,
-                               @Nullable IngesterContext ingesterContext) {
+                               List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportSourceTagIngesterFormatter.java:32]: nit: format

File: java-lib/src/main/java/com/wavefront/ingester/ReportMetricIngesterFormatter.java [hunk 7]
```diff
@@ -35,9 +35,9 @@ public static IngesterFormatBuilder<ReportMetric> newBuilder() {
 
   @Override
   public ReportMetric drive(String input, Supplier<String> defaultHostNameSupplier,
-                           String customerId, @Nullable List<String> customSourceTags,
-                           @Nullable List<String> customLogTimestampTags, @Nullable List<String> customLogMessageTags,
-                           @Nullable IngesterContext ingesterContext) {
+                            String customerId, @Nullable List<String> customSourceTags,
+                            @Nullable List<String> customLogTimestampTags, @Nullable List<String> customLogMessageTags,
+                            List<String> customLogApplicationTags, List<String> customLogServiceTags, @Nullable IngesterContext ingesterContext) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/ReportMetricIngesterFormatter.java:40]: nit: format, lets move ingesterContext to next line

**Review by joannak-vmware — APPROVED** [review]
LGTM, left format nit comments. Ready to merge after reformatted. @xuranchen

**Comment by joannak-vmware** [comment]
Update: Will address formatting in a separate ticket. Merging changes.

**Pull request merged**